src/cargo/ops/cargo_compile.rs

```diff
@@ -494,6 +499,29 @@ pub fn create_bcx<'a, 'cfg>(
         &interner,
     )?;
 
+    if honor_rust_version {
+        let current_version = &target_data.rustc.version;
+        let untagged_version = semver::Version::new(
+            current_version.major,
+            current_version.minor,
+            current_version.patch,
+        );
+
+        for unit in unit_graph.keys() {
+            if let Some(req) = unit.pkg.rust_version() {
+                if !req.matches(&untagged_version) {
+                    anyhow::bail!(
+                        "package `{}` cannot be built because it requires rustc {}, while \
+                         the currently active rustc version is {}",
+                        unit.pkg.name(),
```

Comment:
It might be best to use `unit.pkg`, so if there are multiple instances of the same name, it can be clear which one it is referring to.

Reply:
Fixed.